packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts

```diff
@@ -236,22 +250,32 @@ export class RedirectsMiddleware extends MiddlewareBase {
            * it returns `undefined`. The `matchedQueryString` is later used to indicate whether the query
            * string contributed to a successful redirect match.
            */
-          const matchedQueryString = this.isPermutedQueryMatch({
-            pathname: targetURL,
-            queryString: targetQS,
-            pattern: redirect.pattern,
-            locale,
-          });
+          if (req.headers.get('cdn-loop') === NAME_NETLIFY) {
+            matchedQueryString = this.getPermutedQueryMatch({
+              pathname: normalizedPath,
+              queryString: targetQS,
+              pattern: redirect.pattern,
+              locale,
+            });
+          } else {
+            matchedQueryString = [
```

Comment:
To me looks like this bunch of code does almost the same what `getPermutedQueryMatch` does
do you just need to use the same method?